crates/nu-table/src/style_computer.rs

```diff
@@ -0,0 +1,228 @@
+use crate::TextStyle;
+use nu_ansi_term::{Color, Style};
+use nu_color_config::{color_record_to_nustyle, lookup_ansi_color_style};
+use nu_engine::eval_block;
+use nu_protocol::{
+    engine::{EngineState, Stack},
+    IntoPipelineData, Value,
+};
+use tabled::alignment::AlignmentHorizontal;
+
+use std::collections::HashMap;
+
+// ComputableStyle represents the valid user style types: a single color value, or a closure which
+// takes an input value and produces a color value. The latter represents a value which
+// is computed at use-time.
+#[derive(Clone)]
+pub enum ComputableStyle {
+    Static(Style),
+    Closure(Value),
+}
+
+// macro used for adding initial values to the style hashmap
+macro_rules! initial {
+    ($a:expr, $b:expr) => {
+        ($a.to_string(), ComputableStyle::Static($b))
+    };
+}
+
+// An alias for the mapping used internally by StyleComputer.
+pub type StyleMapping = HashMap<String, ComputableStyle>;
+//
+// A StyleComputer is an all-in-one way to compute styles. A nu command can
+// simply create it with from_config(), and then use it with compute().
+// It stores the engine state and stack needed to run closures that
+// may be defined as a user style.
+//
+pub struct StyleComputer<'a> {
```

Comment:
Do you think it belong to `nu_table`?
I mean it could be used by many, no only for table construction, in which case there's no reason to depend on `nu_table`.